osu.Desktop.VisualTests/Tests/TestCaseBeatmapDetails.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Testing;
+using osu.Game.Database;
+using osu.Game.Screens.Select;
+using System;
+using System.Linq;
+
+namespace osu.Desktop.VisualTests.Tests
+{
+    internal class TestCaseBeatmapDetails : TestCase
+    {
+        public override string Description => "BeatmapDetails tab of BeatmapDetailArea";
+
+        private BeatmapDetails details;
+
+        public override void Reset()
+        {
+            base.Reset();
+
+            Add(details = new BeatmapDetails
+            {
+                RelativeSizeAxes = Axes.Both,
+                Padding = new MarginPadding(150),
+                Beatmap = new BeatmapInfo
+                {
+                    Version = "VisualTest",
+                    Metadata = new BeatmapMetadata
+                    {
+                        Source = "Some guy",
+                        Tags = "beatmap metadata example with a very very long list of tags and not much creativity",
+                    },
+                    Difficulty = new BeatmapDifficulty
+                    {
+                        CircleSize = 7,
+                        ApproachRate = 3.5f,
+                        OverallDifficulty = 5.7f,
+                        DrainRate = 1,
+                    },
+                    StarDifficulty = 5.3f,
+                },
+            });
+
+            AddRepeatStep("new fail values", newRetryAndFailValues, 10);
+            AddStep("new ratings", () => details.Ratings = Enumerable.Range(1, 10));
+            AddStep("remove fails", () => details.Fails = null );

```

Comment:
once you've made everything dependent on `BeatmapInfo` these won't be relevant, but for future reference try not to hide data as the last steps of a testcase. they leave the end-user in a bad state to actually test things out.